ipaserver/plugins/cert.py

```diff
@@ -858,11 +861,16 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
                             "subject alt name type %s is forbidden "
                             "for non-user principals") % "RFC822Name"
                     )
+            elif isinstance(gn, cryptography.x509.general_name.IPAddress):
+                san_has_ipaddrs = True
             else:
                 raise errors.ACIError(
                     info=_("Subject alt name type %s is forbidden")
                     % type(gn).__name__)
 
+        if san_has_ipaddrs:
+            _validate_san_ips(generalnames)
```

Comment:
The ``validate_san_ips`` function iterates over general names again to construct a set of SAN dNSNames and IPAddresses. It would be easier to create both sets ``san_ipaddrs`` and ``san_dnsnames`` in this method and replace ``if san_has_ipaddrs`` with ``if san_ipaddrs``.

Reply:
Haha, @tiran and I were both reviewing the PR at the same time and came up with mostly the same comments ^_^